sql/src/main/java/io/crate/analyze/QueriedSelectRelation.java

```diff
@@ -143,11 +143,22 @@ public boolean hasAggregates() {
         return querySpec.hasAggregates();
     }
 
+    /**
+     * Creates a new relation with the newSubrelation as child.
+     * Fields will be re-mapped (They contain a hard-reference to a relation),
+     * but for this to work the new relation must have semantically equal outputs to the old relation.
+     */
     public AnalyzedRelation replaceSubRelation(AnalyzedRelation newSubRelation) {
         var mapFieldsToNewRelation = FieldReplacer.bind(
-            f -> f.relation().equals(subRelation)
-                ? newSubRelation.getField(f.path(), Operation.READ)
-                : f
+            f -> {
+                if (f.relation().equals(subRelation)) {
+                    int idx = subRelation.fields().indexOf(f);
```

Comment:
shouldn't this use `newSubRelation`?

Reply:
It looks up the position of the old field in the old relation and returns the field at the same position in the new relation.